pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -1311,10 +1501,27 @@ func (nc *Controller) cancelPodEviction(node *v1.Node) bool {
 
 // evictPods queues an eviction for the provided node name, and returns false if the node is already
 // queued for eviction.
-func (nc *Controller) evictPods(node *v1.Node) bool {
+// If node was already evicted, the pods are deleted immediately.
+func (nc *Controller) evictPods(node *v1.Node, pods []*v1.Pod) (bool, error) {
 	nc.evictorLock.Lock()
 	defer nc.evictorLock.Unlock()
-	return nc.zonePodEvictor[utilnode.GetZoneKey(node)].Add(node.Name, string(node.UID))
+	status, ok := nc.nodeEvictionMap.getStatus(node.Name)
+	if ok && status == evicted {
+		// Node eviction already happened for this node.
+		// Handling immediate pod deletion.
+		_, err := nodeutil.DeletePods(nc.kubeClient, pods, nc.recorder, node.Name, string(node.UID), nc.daemonSetStore)
+		if err != nil {
+			return true, fmt.Errorf("unable to delete pods from node %q: %v", node.Name, err)
+		}
+		return true, nil
```

Comment:
This seems wrong - see how the result is handled. e.g. you log
"Node is NotReady. Adding Pods on Node %s to eviction queu..."

that's not what is happening in this path.

The logic and the comment should be changed so that it actually makes sense. (maybe bool is actually not what it should return?)

Reply:
Done.